src/Tasks.UnitTests/Copy_Tests.cs

```diff
@@ -2048,52 +2165,47 @@ public void CopyToDestinationFolderWithHardLinkFallbackNetwork()
-                Assert.True(File.Exists(destFile)); // "destination exists"
+                Assert.True(File.Exists(destFile1)); // "destination exists"
+                Assert.True(File.Exists(destFile2)); // "destination exists"
                 MockEngine.GetStringDelegate resourceDelegate = AssemblyResources.GetString;
 
-                me.AssertLogContainsMessageFromResource(resourceDelegate, "Copy.HardLinkComment", sourceFile, destFile);
+                me.AssertLogContainsMessageFromResource(resourceDelegate, "Copy.HardLinkComment", sourceFile1, destFile1);
 
                 // Can't do this below, because the real message doesn't end with String.Empty, it ends with a CLR exception string, and so matching breaks in PLOC.
-                // Instead look for the HRESULT that CLR unfortunately puts inside its exception string. Something like this
-                // The system cannot move the file to a different disk drive. (Exception from HRESULT: 0x80070011)
+                // Instead look for the HRESULT that CLR unfortunately puts inside its exception string. Something like this:
+                //   The system cannot move the file to a different disk drive. (Exception from HRESULT: 0x80070011)
                 // me.AssertLogContainsMessageFromResource(resourceDelegate, "Copy.RetryingAsFileCopy", sourceFile, destFile, String.Empty);
                 me.AssertLogContains("0x80070011");
 
-                string destinationFileContents;
-                using (StreamReader sr = FileUtilities.OpenRead(destFile))
-                {
-                    destinationFileContents = sr.ReadToEnd();
-                }
-
-                Assert.Equal("This is a source temp file.", destinationFileContents); //"Expected the destination file to contain the contents of source file."
+                string destinationFileContents = File.ReadAllText(destFile1);
+                Assert.Equal("This is source temp file 1.", destinationFileContents); //"Expected the destination file to contain the contents of source file."
+                destinationFileContents = File.ReadAllText(destFile2);
+                Assert.Equal("This is source temp file 2.", destinationFileContents); //"Expected the destination file to contain the contents of source file."
 
-                Assert.Equal(1, t.DestinationFiles.Length);
-                Assert.Equal(1, t.CopiedFiles.Length);
-                Assert.Equal(destFile, t.DestinationFiles[0].ItemSpec);
-                Assert.Equal(destFile, t.CopiedFiles[0].ItemSpec);
+                Assert.Equal(2, t.DestinationFiles.Length);
+                Assert.Equal(2, t.CopiedFiles.Length);
+                Assert.Equal(destFile1, t.DestinationFiles[0].ItemSpec);
+                Assert.Equal(destFile2, t.DestinationFiles[1].ItemSpec);
+                Assert.Equal(destFile1, t.CopiedFiles[0].ItemSpec);
+                Assert.Equal(destFile2, t.CopiedFiles[1].ItemSpec);
```

Comment:
No relevant order problems here?

Reply:
No, in fact if these for some reason came out in a different order it's a bug: These outputs must be ordered in the same order as the inputs as part of the contract. Which is why the parallel code has some complexity in applying logic partitions by destination file path.

Reply:
Makes sense!